app/components/upload-app/status/index.ts

```diff
@@ -0,0 +1,177 @@
+          obj.completedSubmissions = obj.completedSubmissions + 1;
+        } else if (
+          ENUMS.SUBMISSION_STATUS_OBJ.FAILED.includes(submission.status)
+        ) {
+          obj.failedSubmissions = obj.failedSubmissions + 1;
+        } else {
+          obj.runningSubmissions = obj.runningSubmissions + 1;
+        }
+      });
+    }
+    if (this.uploadedApps?.files?.length > 0) {
+      obj.runningSubmissions =
+        obj.runningSubmissions + this.uploadedApps.files.length;
+    }
+    obj.variant = obj.runningSubmissions > 0 ? 'indeterminate' : 'determinate';
+    obj.loaderColor = obj.failedSubmissions > 0 ? '' : 'green-loader';
+    return obj;
+  }
+
+  get uploadedApps() {
+    return this.uploadApp.uploadingApps;
+  }
+
+  @action
+  registerAnchorRef(element: HTMLElement | null) {
+    this.anchorRef = element;
+  }
+
+  @computed('submissionList.@each.createdOn')
+  get sortedSubmissions() {
+    return this.submissionList
+      ?.slice()
+      .sort((a: any, b: any) => b.createdOn - a.createdOn);
+  }
+
+  @computed('submissionList.length')
+  get hasSortedSubmissions() {
+    return this.submissionList.length > 0;
+  }
+
+  @computed('submissions.length')
+  get hasSubmissions() {
+    return this.submissions ? true : false;
+  }
+
+  get submissionListArray() {
+    let failed: SubmissionModel[] = [];
+    let uploading: any = [];
+    let completed: SubmissionModel[] = [];
+    let running: SubmissionModel[] = [];
+    if (this.uploadedApps?.files?.length > 0) {
+      this.uploadedApps.files.map((files: any) => {
+        files.statusPercentage = files.progress;
+        files.statusHumanized = this.intl.t('uploading');
+        uploading.push(files);
+      });
+    }
+    this.sortedSubmissions.map((submission) => {
+      if (submission.status === ENUMS.SUBMISSION_STATUS.ANALYZING) {
+        completed.push(submission);
+      } else if (
+        ENUMS.SUBMISSION_STATUS_OBJ.FAILED.includes(submission.status)
+      ) {
+        failed.push(submission);
+      } else {
+        running.push(submission);
+      }
+    });
+    return failed.concat(uploading, running, completed);
+  }
+
+  @action initialize() {
+    addObserver(
+      this.realtime,
+      'SubmissionCounter',
+      this,
+      this.observeSubmissionCounter
+    );
+
+    this.getSubmissions.perform();
+  }
+
+  @action removeSubmissionCounterObserver() {
+    removeObserver(
+      this.realtime,
+      'SubmissionCounter',
+      this,
+      this.observeSubmissionCounter
+    );
+  }
+
+  @action
+  observeSubmissionCounter() {
+    this.getSubmissions.perform();
+  }
+
+  getSubmissions = task(async () => {
+    try {
+      this.store
+        .query('submission', {
+          status: ENUMS.SUBMISSION_STATUS.VALIDATING,
+        })
+        .then((submissions) => {
+          this.submissions = submissions;
+
+          return (this.submissions = this.store.peekAll('submission'));
```

Comment:
why are we setting `submissions` and again reseting the it with `peekAll` ? and we change this to async await